lib/Sema/CSGen.cpp

```diff
@@ -4396,140 +4395,196 @@ generateForEachStmtConstraints(ConstraintSystem &cs,
 
   auto *makeIteratorVar = new (ctx)
       VarDecl(/*isStatic=*/false, VarDecl::Introducer::Var,
-              sequenceExpr->getStartLoc(), ctx.getIdentifier(name), dc);
+              forEachExpr->getStartLoc(), ctx.getIdentifier(name), dc);
   makeIteratorVar->setImplicit();
 
-  // First, let's form a call from sequence to `.makeIterator()` and save
-  // that in a special variable which is going to be used by SILGen.
-  {
-    FuncDecl *makeIterator = isAsync ? ctx.getAsyncSequenceMakeAsyncIterator()
-                                     : ctx.getSequenceMakeIterator();
+    // Retrieve the pattern type and generate constraints when dealing with a pack
+    // expansion in a for-in loop.
+  if (ctx.LangOpts.hasFeature(Feature::PackIteration) &&
+      isa<PackExpansionExpr>(forEachExpr)) {
```

Comment:
If the feature is not enabled we need to produce a fix that would diagnose that fact in `generateForEachStmtConstraints` and probably skip the reset of the work for the statement because it's invalid instead of falling back the original logic

Reply:
Note that having this new fix is the most important bit here for user experience.